lib/config/genconfig/hadeployconfig.go

```diff
@@ -0,0 +1,1102 @@
+	ip, err1 := c.PromptNodeIp(fmt.Sprintf(PROMPT_NODE_IP, i+1))
+	if err1 != nil {
+		return err1
+	}
+	c.Config.ExistingInfra.Config.OpensearchPrivateIps = append(c.Config.ExistingInfra.Config.OpensearchPrivateIps, ip)
+	if hasCustomCerts && hasCustomCertsPerNode {
+		priKey, err := c.PromptCert(ip, OPENSEARCH_NODETYPE, PRI_KEY)
+		if err != nil {
+			return err
+		}
+
+		pubCert, err := c.PromptCert(ip, OPENSEARCH_NODETYPE, PUB_CERT)
+		if err != nil {
+			return err
+		}
+
+		nodeCert := config.CertByIP{
+			IP:         ip,
+			PrivateKey: priKey,
+			PublicKey:  pubCert,
+		}
+		*c.Config.Opensearch.Config.CertsByIP = append(*c.Config.Opensearch.Config.CertsByIP, nodeCert)
+	}
+	return
+}
+
+func (haconfiggen *HaDeployConfigGen) PromptOpenSearchPubPriCerts(hasCustomCerts, hasCustomCertsPerNode bool) (err error) {
+	if hasCustomCerts && !hasCustomCertsPerNode {
+		priKey, err := haconfiggen.PromptCert("", OPENSEARCH_NODETYPE, PRI_KEY)
+		if err != nil {
+			return err
+		}
+		haconfiggen.Config.InitOpenSearch().InitConfig().PrivateKey = priKey
+
+		pubCert, err := haconfiggen.PromptCert("", OPENSEARCH_NODETYPE, PUB_CERT)
+		if err != nil {
+			return err
+		}
+		haconfiggen.Config.InitOpenSearch().InitConfig().PublicKey = pubCert
+	}
+
+	if hasCustomCerts {
+		rootCa, err := haconfiggen.PromptCert("", OPENSEARCH_NODETYPE, ROOTCA)
+		if err != nil {
+			return err
+		}
+		haconfiggen.Config.Opensearch.Config.RootCA = rootCa
+		adminCert, err := haconfiggen.PromptCert("", OPENSEARCH_NODETYPE, ADMIN_CERT)
+		if err != nil {
+			return err
+		}
+		haconfiggen.Config.Opensearch.Config.AdminCert = adminCert
+		adminKey, err := haconfiggen.PromptCert("", OPENSEARCH_NODETYPE, ADMIN_KEY)
+		if err != nil {
+			return err
+		}
+		haconfiggen.Config.Opensearch.Config.AdminKey = adminKey
+	}
+
+	if hasCustomCerts && hasCustomCertsPerNode {
+		haconfiggen.Config.Opensearch.Config.InitCertsByIP()
+	}
+	return
+}
+
+func (c *HaDeployConfigGen) PromptPostgresqlNodes() (err error) {
+	postgresqlNodeCount, err := c.Prompt.InputIntRange("No. of Postgresql Nodes", 3, 100)
+	if err != nil {
+		return
+	}
+	c.Config.InitPostgresql().InitConfig()
+	c.Config.Postgresql.Config.InstanceCount = fmt.Sprint(postgresqlNodeCount)
+
+	hasCustomCerts, hasCustomCertsPerNode, err := c.PromptHaveCustomCerts(POSTGRESQL_NODETYPE)
+	if err != nil {
+		return err
+	}
+
+	c.Config.Postgresql.Config.EnableCustomCerts = hasCustomCerts
+
+	err = c.PromptPostgresqlPubPriCerts(hasCustomCerts, hasCustomCertsPerNode)
+	if err != nil {
+		return err
+	}
+
+	c.Config.InitExistingInfra().InitConfig()
+	for i := 0; i < postgresqlNodeCount; i++ {
+		err = c.PromptPostgresqlIpCerts(i, hasCustomCerts, hasCustomCertsPerNode)
+		if err != nil {
+			return err
+		}
+	}
+	return
+}
+
+func (c *HaDeployConfigGen) PromptPostgresqlIpCerts(i int, hasCustomCerts, hasCustomCertsPerNode bool) (err error) {
+	ip, err1 := c.PromptNodeIp(fmt.Sprintf(PROMPT_NODE_IP, i+1))
```

Comment:
in case of more than 1 node count, the IPs should be unique, we can add checks for that

Reply:
I will consider this as feature enhancement for next time.